lib/spack/spack/compilers/msvc.py

```diff
@@ -115,15 +107,16 @@ def command_str(self):
         return f"{script} {self.arch} {self.sdk_ver} {self.vcvars_ver}"
 
 
-def get_valid_fortran_pth(comp_ver):
-    cl_ver = str(comp_ver)
-    sort_fn = lambda fc_ver: Version(fc_ver)
-    sort_fc_ver = sorted(list(avail_fc_version), key=sort_fn)
-    for ver in sort_fc_ver:
-        if ver in fortran_mapping:
-            if Version(cl_ver) <= Version(fortran_mapping[ver]):
-                return fc_path[ver]
-    return None
+def get_valid_fortran_pth():
+    """Assign maximum available fortran compiler version"""
+    # TODO (johnwparent): validate compatibility w/ try compiler
+    # functionality when added
+    valid_fortran_path = None
+    if FC_PATH.get(""):
```

Comment:
I'm having trouble seeing why you'd be looking for `""` or how that gets set: would it make more sense to always iterate over the keys? Or is something invalid about them until the `""` key is set? If it's possible to do this without setting `""` as a key I think it would be clearer i.e. doing

```
if fc_ver:
    FC_PATH[fc_ver] = fc
```

below

Reply:
I think the intent here was, based on your comment suggesting to use `FC_PATH.get(..., None)`, to just check for the presence of any fortran compilers and I got interrupted and didn't notice the typo. 
Refactored and pushed.